### PR TITLE
Merge : TDDFT branch

### PR DESCRIPTION
**Pull request by lyb9812**
1. update fixed tddft codes in TDDFT branch
2. apply new hamiltonian hsolver elecstate in tddft module
3. fix serial version

### CODE REVIEW

File: source/module_elecstate/elecstate_lcao_tddft.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                                       this->lowf->wfc_k_grid[ik],
+                                       ik,
+                                       this->ekb,
+                                       this->wg);
+#else
+            for (int ib = 0; ib < GlobalV::NBANDS; ib++)
+            {
+                for (int iw = 0; iw < GlobalV::NLOCAL; iw++)
+                {
+                    this->lowf->wfc_k_grid[ik][ib][iw] = psi.get_pointer()[ib * GlobalV::NLOCAL + iw];
```
Comment (dyzheng) [inline source/module_elecstate/elecstate_lcao_tddft.cpp]: Why don't you use "psi(ib, iw)" there?

File: source/module_elecstate/elecstate_lcao_tddft.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    ModuleBase::timer::tick("ElecStateLCAO", "psiToRho");
+    return;
+}
+
+void ElecStateLCAO_TDDFT::calculate_weights_td()
+{
+    ModuleBase::TITLE("ElecState", "calculate_weights");
+
+    if (GlobalV::ocp == 1)
```
Comment (dyzheng) [inline source/module_elecstate/elecstate_lcao_tddft.cpp:68]: Does user-defined occupation only can be used in TDDFT?

File: source/module_esolver/esolver_ks.cpp [hunk 3]
```diff
@@ -4,6 +4,8 @@
 #include "../src_io/print_info.h"
 #ifdef __MPI
 #include "mpi.h"
+#else
+#include "chrono"
```
Comment (dyzheng) [inline source/module_esolver/esolver_ks.cpp:8]: why do you add this header here?
There is no addition code in this cpp file without this header.

File: source/module_esolver/esolver_ks_lcao.cpp [hunk 4]
```diff
@@ -415,14 +415,9 @@ namespace ModuleESolver
                 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                 GlobalC::pot.vr = GlobalC::pot.v_of_rho(GlobalC::CHR.rho, GlobalC::CHR.rho_core);
                 GlobalC::en.delta_escf();
-                if (ELEC_evolve::td_vext == 0)
-                {
+
```
Comment (dyzheng) [inline source/module_esolver/esolver_ks_lcao.cpp]: Please 1. keep the code aligned and 2. add some annotation here to explain what will next code do.

File: source/src_pw/potential.h [hunk 5]
```diff
@@ -1,94 +1,91 @@
...
 
+#include "../module_base/complexmatrix.h"
 #include "../module_base/global_function.h"
 #include "../module_base/global_variable.h"
 #include "../module_base/matrix.h"
-#include "../module_base/complexmatrix.h"
-#include "module_esolver/esolver_ks_lcao.h"
 #include "../module_pw/pw_basis.h"
+#include "module_esolver/esolver_ks_lcao.h"
+#include "module_esolver/esolver_ks_lcao_tddft.h"
```
Comment (dyzheng) [inline source/src_pw/potential.h]: There is no need to add these two headers here?

**Review by dyzheng — CHANGES_REQUESTED** [review]
What did these three files do:
"source/src_lcao/ELEC_cbands_gamma.h"
"source/src_lcao/ELEC_cbands_k.h"
"source/src_lcao/ELEC_scf.cpp"
These old files has been replaced to HSolver and ESolver.

File: source/module_elecstate/elecstate_lcao_tddft.cpp (hunk 1 above)
Comment (lyb9812) [reply source/module_elecstate/elecstate_lcao_tddft.cpp]: ok, I will replace it

File: source/module_elecstate/elecstate_lcao_tddft.cpp (hunk 2 above)
Comment (lyb9812) [reply source/module_elecstate/elecstate_lcao_tddft.cpp:68]: Currently it is only used in TDDFT, and it need to be used after the first step (ground state).

File: source/module_esolver/esolver_ks.cpp (hunk 3 above)
Comment (lyb9812) [reply source/module_esolver/esolver_ks.cpp:8]: The serial code line 185, 243 need this header.

File: source/src_pw/potential.h (hunk 5 above)
Comment (lyb9812) [reply source/src_pw/potential.h]: Yes, I will delete them.

**Comment by caic99** [comment]
@dyzheng Please ignore the static analysis test, this is an external bug.

**Comment by dyzheng** [comment]
> What did these three files do: "source/src_lcao/ELEC_cbands_gamma.h" "source/src_lcao/ELEC_cbands_k.h" "source/src_lcao/ELEC_scf.cpp" These old files has been replaced to HSolver and ESolver.

What usage of these three new added files?@lyb9812

File: source/src_pw/potential.h (hunk 5 above)
Comment (lyb9812) [reply source/src_pw/potential.h]: I delete friend class solver_ks_lcao and move set_vrs_tddft from private to public now.

**Comment by lyb9812** [comment]
> 

I have already delete them now.

**Comment by dyzheng** [comment]
There is still conflict in FORCE_k.cpp, maybe caused by Meta_GGA in LCAO, please solve it.

**Review by dyzheng — APPROVED** [review]

**Pull request merged**